### PR TITLE
Moved docs tree generation into docsTree.ts

### PR DESCRIPTION
**Pull request by oleklukasiewicz**
## Description
Moved docs tree generation from "src\routes\docs\+layout.server.ts" into "src\data\docsTree.ts".

## Motivation and Context
Before this change, docs tree was only generated when you went to the documentation page, which caused a problem #339.

### CODE REVIEW

File: src/data/docsTree.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const getTree = (pages: DocsNode[]) => {
+    const rawCategories = import.meta.glob<DocsMetadata>("../routes/docs/**/*/category.md", {
+        eager: true,
+        import: "metadata"
+    });
+
+    const categories: DocsTree = Object.entries(rawCategories).map(
+        ([path, node]) => {
+            const docsPath =
+                path.match(PATH_TRIM)?.[1] ?? "this is an invalid path";
```
Comment (TheOnlyTails) [inline src/data/docsTree.ts:30]: It's better to throw an error here instead of moving forward with potentially corrupt data.
```suggestion
                path.match(PATH_TRIM)?.[1] ?? throw Error(`this is an invalid docs page path: $path`);
```

File: src/routes/+layout.svelte [hunk 2]
```diff
@@ -1,22 +1,17 @@
 <script lang="ts">
 	import { dev } from "$app/environment";
 	import { page } from "$app/stores";
-
```
Comment (TheOnlyTails) [inline src/routes/+layout.svelte:4]: Why did you remove all the blank lines in this file? Please add them back. all formatting in this project should be done according to the rules defined in the prettier config.

File: src/routes/+layout.ts [hunk 3]
```diff
@@ -1,12 +1,17 @@
 import type { LayoutLoad } from "./$types";
+import { docsPages, docsTree } from "$data/docsTree";
 import { waitLocale } from "svelte-i18n";
 import i18n from "$i18n";
 
-export const load: LayoutLoad = () => {
+export const load: LayoutLoad = ({ url, route }) => {
```
Comment (TheOnlyTails) [inline src/routes/+layout.ts:6]: why are you including the `url` and `route` params here if they're not used?

File: src/routes/docs/+layout.server.ts [hunk 4]
```diff
@@ -1,57 +1,13 @@
...
-	const tree = getTree(pages);
 
 	return {
 		pagePath: url.pathname,
-		currentPage: pages.find(
+		currentPage: docsPages.find(
 			p => p.path === route.id.substring(route.id.indexOf("/", 2))
 		) ?? { title: "Overview", path: "" },
-		docsPages: pages,
-		docs: tree
```
Comment (TheOnlyTails) [inline src/routes/docs/+layout.server.ts:55]: I believe removing these 2 lines will break the `data` prop in https://github.com/files-community/Website/blob/main/src/routes/docs/+layout.svelte#L13. You can forward them from the parent layout load using `await parent()`.

**Review by TheOnlyTails — CHANGES_REQUESTED** [review]
This is a good idea, but needs some improvements.

File: src/data/docsTree.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import type { DocsMetadata, DocsNode, DocsTree } from "$data/docs";
+
+const PATH_TRIM = /(?:..\/routes\/docs)((\/[\w-]+)+)\//;
+
+const getPages = () => {
+    console.log("SEARH")
+    console.trace();
```
[the remark this answers is not in the document]
Comment (oleklukasiewicz) [reply src/data/docsTree.ts:7]: ok, sorry :P

File: src/data/docsTree.ts (hunk 1 above)
Comment (oleklukasiewicz) [reply src/data/docsTree.ts:30]: ok

File: src/routes/+layout.svelte (hunk 2 above)
Comment (oleklukasiewicz) [reply src/routes/+layout.svelte:4]: ok

File: src/routes/docs/+layout.server.ts (hunk 4 above)
Comment (oleklukasiewicz) [reply src/routes/docs/+layout.server.ts:55]: When I checked parent was an empty object, so I think it's better to just get the pages tree from "docsTree.ts".

File: src/data/docsTree.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    const rawCategories = import.meta.glob<DocsMetadata>("../routes/docs/**/*/category.md", {
+        eager: true,
+        import: "metadata"
+    });
+
+    const categories: DocsTree = Object.entries(rawCategories).map(
+        ([path, node]) => {
+            const docsPath = path.match(PATH_TRIM)?.[1];
+            if (!docsPath)
+                throw new Error(`this is an invalid docs page path: $path`);
```
Comment (TheOnlyTails) [inline src/data/docsTree.ts:29]: this is my bad, we should be throwing SvelteKit's `error` function here: https://kit.svelte.dev/docs/load#errors

Comment (oleklukasiewicz) [reply src/data/docsTree.ts]: ok, changed!

File: src/data/docsTree.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    const rawCategories = import.meta.glob<DocsMetadata>("../routes/docs/**/*/category.md", {
+        eager: true,
+        import: "metadata"
+    });
+
+    const categories: DocsTree = Object.entries(rawCategories).map(
+        ([path, node]) => {
+            const docsPath = path.match(PATH_TRIM)?.[1];
+            if (!docsPath)
+                throw error(404, `this is an invalid docs page path: $path`);
```
Comment (TheOnlyTails) [inline src/data/docsTree.ts:30]:
```suggestion
                throw error(500, `this is an invalid docs page path: $path`);
```

**Review by TheOnlyTails — APPROVED** [review]

**Comment by yaira2** [comment]
@oleklukasiewicz thank you
